### PR TITLE
Add integration test for Azure multi block uploads

### PR DESCRIPTION
**Pull request by fcofdez**
Relates to https://github.com/elastic/elasticsearch/pull/68957

### CODE REVIEW

**Comment by fcofdez** [comment]
@original-brownbear  would you mind reviewing this PR when you have the chance? Thanks!

File: plugins/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureStorageCleanupThirdPartyTests.java [hunk 1]
```diff
@@ -111,4 +117,19 @@ private void ensureSasTokenPermissions() {
...
         future.actionGet();
     }
+
+    public void testMultiBlockUpload() throws Exception {
+        final BlobStoreRepository repo = getRepository();
+        // The configured threshold for this test suite is 1mb
+        final int blobSize = ByteSizeUnit.MB.toIntBytes(2);
+        final Executor executor = repo.threadPool().generic();
+        PlainActionFuture<Void> future = PlainActionFuture.newFuture();
+        executor.execute(ActionRunnable.run(future, () -> {
```
Comment (original-brownbear) [inline plugins/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureStorageCleanupThirdPartyTests.java:127]: NIT: just inline `repo.threadPool().generic()` here, you're only using `executor` once

**Review by original-brownbear — APPROVED** [review]
LGTM :) + sorry for the delay!

**Comment by fcofdez** [comment]
@elasticmachine update branch

**Pull request merged**